INTER-222-HeaderComponent-공용 컴포넌트 헤더 구현

mulgko opened the pull request:
## PR 본문

### 반영 브랜치

INTER-222-HeaderComponent -> dev

### PR 타입(하나 이상의 PR 타입을 선택해주세요)

- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 작업 내용

- 이번 PR에서 작업한 내용을 간략히 설명해주세요. (이미지 첨부 가능)
- 헤더 컴포넌트 구현 

### 스크린샷 (선택)
<img width="1488" alt="image" src="https://github.com/user-attachments/assets/af5596f7-7bf6-465f-93cc-4a0c7fa1eaa8" />

https://github.com/user-attachments/assets/7c1e926b-4411-4e10-80af-696f591b0599

mulgko committed:
feat: header base
nav
sidebar
overlay

mulgko committed:
feat: header 기능 추가
search modal base

mulgko committed:
feat: header에 dev내용추가
tailwind 변수 추가
config 내용 삭제

mulgko committed:
Merge branch 'dev' into INTER-222-HeaderComponent

mulgko committed:
style: pretandard 적용

mulgko committed:
test: header test

mulgko committed:
style: header icon size 변경

mulgko committed:
merge: dev ᆺ브랜치 병합

mulgko committed:
merge: dev브랜치와 병합

mulgko committed:
feat: 드롭다운 추가
- 게임 만들기
- 글 올리기
- ICON 추가
- text에서 font로 변수명 변경

mulgko committed:
fix: 아이콘 버튼 추가
- createNew 버튼 추가
- Login 버튼 추가

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+              onClick={() => setSearchModalOpen(true)}
+            >
+              <Button type="button" className={`${buttonVariants({ color: "white" })} !border-0`}>
+                <Icon name="search" className="w-[26px] h-[26px] md:w-[33px] md:h-[33px]" />
```
px단위에 대한 arbitrary class는 되도록 사용하지 않습니다. w-6이나, w-7 / h-6, h-7과 같은 방법으로 수정해주세요. tailwind는 rem기반으로 움직입니다. 단위 일관성을 위해 수정 부탁드립니다.

GiToon10100011 commented on src/app/components/header/Header.tsx:
스크롤 하면 헤더가 접히는 로직도 있으면 좋을 것 같습니다

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+                <div className="mt-4">
+                  <h3 className="text-sm font-medium text-gray-500 mb-2"> 검색어</h3>
+                  <div className="flex flex-wrap gap-2">
+                    <button className="px-3 py-1 bg-gray-100 rounded-full text-sm hover:bg-gray-200 transition-colors">
```
Button 컴포넌트를 사용해주세요.

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+        </div>
+      </div>
+
+      {/* 검색 모달 (검색 버튼 클릭 시 표시) */}
```
검색 모달은 페이지 컴포넌트로 분리될 예정입니다. (추후에 분리 필요)

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+              // 로그인 된 상태 - 아이콘 두 개 추가
+              <>
+                <div className="relative create-menu-container">
+                  <button onClick={() => setCreateMenuOpen(!createMenuOpen)}>
```
마찬가지로 Button 컴포넌트 사용 부탁드립니다.

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+                    className="p-1 hover:bg-gray-100 rounded-full transition-colors"
+                    onClick={() => setSearchModalOpen(false)}
+                  >
+                    <svg
```
svg는 Icon컴포넌트로 사용부탁드립니다.

GiToon10100011 commented on src/app/components/header/header.type.ts:
```diff
@@ -0,0 +1,12 @@
+
+// 헤더 컴포넌트 props 인터페이스
+export interface IHeaderProps {
+  isLoggedIn?: boolean;
```
전역상태를 통해 인증관리가 진행됨으로 인해 추후에 해당 인터페이스도 필요가 없어질 것 같습니다. (추후 변경, 일단 병합)

GiToon10100011 commented on src/app/components/header/header.variants.ts:
```diff
@@ -0,0 +1,83 @@
+      default: "mx-auto px-4 py-4",
+    },
+  },
+  defaultVariants: {
```
variants에 size만 있는데 불필요한 defaultVariants사용으로 보입니다.

GiToon10100011 commented on src/app/components/header/header.variants.ts:
navMenu, createMenu등은 headerVariants내의 variants로 사용해도 되지 않을까요? 리펙토링이 필요할 것 같습니다..! 또한 Button Variants들은 전부 Button 컴포넌트에 이미 정의가 되어있을거라고 판단되어 해당 variants도 필요없을 것 같습니다.

GiToon10100011 replied:
제가 의도 했던건 

```ts
export const headerVariants = tv({
 slots: {
  base: "border-b border-gray-200",
  navMenu: ""font-regular-24 hover:text-main hover:font-sb-24 transition-all",
  createMenu: "absolute right-0 top-full mt-2 w-40 bg-white border border-gray-200 rounded-md shadow-lg z-50 transform origin-top-right transition-all duration-200",
// ...이하
}

 //..variants필요하면 추가
});
```

와 같은 방식으로 리펙토링을 하여, import할거를 최소화 시키고, tailwind-variants가 의도하는 컴포넌트 쪼개는 패턴에 더 맞게 수정을 요청드린거였습니다.  또한 저번 리뷰때 버튼은 어차피 Button컴포넌트에서 variants가 정의되었기에 이에 대한 button 스타일링이 header variants에 필요없을 것 같다고 했는데 유지시킨 이유가 따로 있을까요?

GiToon10100011 commented on src/app/layout.tsx:
RootLayout에서는 헤더를 넣으면 안됩니다. 그룹 라우팅으로 헤더가 있고 없는 페이지들을 정의했기때문에 그룹 내의 layout.tsx에서 사용될 예정입니다. 테스트는 오직 테스트 페이지에서 부탁드립니다.

GiToon10100011 reviewed (changes requested):
고생많으셨습니다. 요청사항 수정 후 재 커밋 부탁드리겠습니다!

mulgko committed:
refactor: fill props 추가 #ᆻ20

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -0,0 +1,315 @@
+  );
+};
+
+export default Header;
```
page.tsx와 같은 테스트 페이지말고는 index.ts를 통한 재 export가 이루어지기 때문에 export default가 아닌 export const Header로 수정 부탁드립니다.

GiToon10100011 commented on src/app/components/header/header.type.ts:
```diff
@@ -0,0 +1,12 @@
+  isLoggedIn?: boolean;
+}
+
+export default IHeaderProps;
```
여기 부분도 제거 부탁드립니다.

GiToon10100011 reviewed (changes requested):
추가 수정 부탁드립니다

mulgko committed:
refactor: 코드 리팩토링
- px 단위 수정
- button component 사용
- svg => icon 사용
- defaultVarinats 삭제
- RootLayout => Header 제거

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -119,27 +117,28 @@ const Header = ({ isLoggedIn }: IHeaderProps) => {
 
           {/* 헤더 우측: 검색 버튼, 로그인 */}
-          <div className="flex items-center gap-[12px] md:gap-[29px]">
+          <div className="flex items-center gap-ᆻ md:gap-6">
```
gap 오타가 난것으로 보입니다.

GiToon10100011 commented on src/app/components/header/Header.tsx:
```diff
@@ -160,19 +159,14 @@ const Header = ({ isLoggedIn }: IHeaderProps) => {
-                    name="setting"
-                    className="w-[26px] h-[26px] md:w-[33px] md:h-[33px] cursor-pointer"
-                  />
+                  <Icon name="setting" className="w-6] h-6 md:w-8 md:h-8 cursor-pointer" />
```
대괄호 오타가 있는것으로 보입니다.

mulgko committed:
refactor: 코드리팩토링

mulgko committed:
refactor: 코드 리팩토링

mulgko committed:
Merge remote-tracking branch 'origin/ISSUE-26-ButtonDefaultDelete' into INTER-222-HeaderComponent

mulgko committed:
refactor: variants 수정
- header, navMenu, createMenu, login / type으로 정의

GiToon10100011 commented on src/app/components/header/header.type.ts:
```diff
@@ -0,0 +1,12 @@
+  isLoggedIn?: boolean;
+}
+
+export default IHeaderProps;
```
이 부분도 제거 부탁드립니다.

mulgko replied:
넵 확인하느라 고생이 많으십니다 화이팅

GiToon10100011 reviewed (changes requested):
수정하느라 고생이 많으십니다. 파이팅!

mulgko committed:
refactor: interface 제거

mulgko committed:
refactor: variants slot 사용

GiToon10100011 commented:
확인해보니 헤더가 position이 fixed, sticky 어느것도 아니어서 스크롤시 안 따라오던데 포지션 수정도 부탁드리겠습니다! 스크롤 시 헤더 접히는걸 추천드리긴했는데 이에 대해 어떻게 생각하시는지에 대한 의견도 함께 물어보고싶습니다

기타 수정사항들은 전부 확인했습니다. 수고하셨습니다.

mulgko committed:
feat: 헤더 접히는 기능 추가

mulgko committed:
refactor: search modal 삭제

mulgko committed:
style: nav 메뉴 폰트 변경
- regular 24 => regular 18로 변경

GiToon10100011 commented:
사이드 메뉴 요소를 더 넣어보고 테스트 결과, 현재 사이드 메뉴 y 스크롤이 발생하고 있지 않습니다. 또한, 사이드바 메뉴의 폰트사이즈도 조금 줄여주시면 감사하겠습니다! 모바일은 수정사항 없어보입니다.

GiToon10100011 commented:
사이드 메뉴 요소를 더 넣어보고 테스트 결과, 현재 사이드 메뉴 y 스크롤이 발생하고 있지 않습니다. 또한, 사이드바 메뉴의 폰트사이즈도 조금 줄여주시면 감사하겠습니다! 모바일은 수정사항 없어보입니다.

mulgko commented:
> 사이드 메뉴 요소를 더 넣어보고 테스트 결과, 현재 사이드 메뉴 y 스크롤이 발생하고 있지 않습니다. 또한, 사이드바 메뉴의 폰트사이즈도 조금 줄여주시면 감사하겠습니다! 모바일은 수정사항 없어보입니다.

아래와 같이 수정하였습니다! y-scroll과 font크기 조정하였습니다
<img width="286" alt="image" src="https://github.com/user-attachments/assets/fe565061-a4aa-4bb1-94fc-c52834f971fc" />

mulgko committed:
refactor: sidebar 폰트 크기 조절, scroll 영억 생성
- scroll-y
- regular-24 => regular-18

GiToon10100011 commented:
LGTM!
